contracts/StakingDistributor.sol

```diff
@@ -1,211 +1,200 @@
-        require(_recipient != address(0), "Zero address: Recipient");
-        require(_rewardRate <= rateDenominator, "Rate cannot exceed denominator");
-        info.push(Info({recipient: _recipient, rate: _rewardRate}));
+    /// @notice sets the liquidity pools for mint and sync
+    /// @dev    note that this overwrites the entire list (!!)
+    function setPools(address[] calldata _pools) external onlyGovernor {
+        pools = _pools;
+    }
+
+    /// @notice removes a pool from the list
+    function removePool(uint256 index, address pool) external onlyGovernor {
+        if (pools[index] != pool) revert Sanity_Check();
+        pools[index] = address(0);
     }
 
-    /**
-        @notice removes recipient for distributions
-        @param _index uint
-     */
-    function removeRecipient(uint256 _index) external override {
-        require(
-            msg.sender == authority.governor() || msg.sender == authority.guardian(),
-            "Caller is not governor or guardian"
-        );
-        require(info[_index].recipient != address(0), "Recipient does not exist");
-        info[_index].recipient = address(0);
-        info[_index].rate = 0;
+    /// @notice adds a pool to the list
+    /// @dev    note you should find an empty slot offchain before calling
+    /// @dev    if there are no empty slots, pass in an occupied index to push
+    function addPool(uint256 index, address pool) external onlyGovernor {
+        // we want to overwrite slots where possible
+        if (pools[index] == address(0)) {
+            pools[index] = pool;
+        } else {
+            // if the passed in slot is not empty, push to the end
+            pools.push(pool);
+        }
     }
 
-    /**
-        @notice set adjustment info for a collector's reward rate
-        @param _index uint
-        @param _add bool
-        @param _rate uint
-        @param _target uint
-     */
+    /// @notice set adjustment info for a collector's reward rate
     function setAdjustment(
-        uint256 _index,
         bool _add,
         uint256 _rate,
         uint256 _target
-    ) external override {
-        require(
-            msg.sender == authority.governor() || msg.sender == authority.guardian(),
-            "Caller is not governor or guardian"
-        );
-        require(info[_index].recipient != address(0), "Recipient does not exist");
-
-        if (msg.sender == authority.guardian()) {
-            require(_rate <= info[_index].rate.mul(25).div(1000), "Limiter: cannot adjust by >2.5%");
-        }
-
-        if (!_add) {
-            require(_rate <= info[_index].rate, "Cannot decrease rate by more than it already is");
-        }
+    ) external {
+        if (msg.sender != authority.governor() && msg.sender != authority.guardian()) revert Not_Permissioned();
+        if (msg.sender == authority.guardian() && _rate > rewardRate * 25 / 1000) revert Adjustment_Limit();
+        if (!_add && _rate > rewardRate) revert Adjustment_Overflow();
```

Comment:
Error would be an underflow instead of an overflow (kind of a nitpick).

Reply:
save it for the semantics dome, E.B. White